src/tests/Interop/PInvoke/Miscellaneous/ThisCall/ThisCallTest.cs

```diff
@@ -40,20 +42,37 @@ public struct IntWrapper
         public int i;
     }
 
+    public enum E : uint
+    {
+        Value = 42
+    }
+
     [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
     public delegate SizeF GetSizeFn(C* c);
     [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
     public delegate Width GetWidthFn(C* c);
     [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
     public delegate IntWrapper GetHeightAsIntFn(C* c);
 
+    [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+    public delegate E GetEFn(C* c);
+
     [DllImport(nameof(ThisCallNative))]
     public static extern C* CreateInstanceOfC(float width, float height);
+
+    [DllImport(nameof(ThisCallNative))]
+    public static extern SizeF GetSizeFromManaged(C* c);
+    [DllImport(nameof(ThisCallNative))]
+    public static extern Width GetWidthFromManaged(C* c);
+    [DllImport(nameof(ThisCallNative))]
+    public static extern IntWrapper GetHeightAsIntFromManaged(C* c);
+    [DllImport(nameof(ThisCallNative))]
+    public static extern E GetEFromManaged(C* c);
 }
 
-class ThisCallTest
+unsafe class ThisCallTest
```

Comment:
Does it makes sense to also create a C++/CLI test for this as well?

Reply:
I think the C++/CLI compiler already accounts for the lack of this PR, so a C++/CLI test wouldn't end up testing what we want.

Reply:
>I think the C++/CLI compiler already accounts for the lack of this PR

Does that mean this change will require a compiler update?

I'm not familiar with all the issues this could create, but given that we are playing with calling conventions C++/CLI does raise some concerns for me given the reliance on it by WPF and other high visibility libraries. I just want to make sure they aren't impact by surprise. That is my main concern with this comment.

Reply:
I'll do a little test with C++/CLI to validate that we won't break anything.